src/bin/pg_autoctl/file_utils.c

```diff
@@ -119,6 +119,24 @@ ensure_empty_dir(const char *dirname, int mode)
 }
 
 
+/*
+ * fopen_with_umask is a version of fopen that gives more control. The main
+ * advantage of it is that it allows specifying a umask of the file. This makes
+ * sure files are not accidentally created with umask 777 if the user has it
+ * configured in a weird way.
+ */
+FILE *
+fopen_with_umask(const char *filePath, const char* modes, int flags, mode_t umask)
+{
+	int fileDescriptor = open(filePath, O_WRONLY | O_CREAT, umask);
+	if (fileDescriptor == -1)
+	{
+		return NULL;
```

Comment:
A comment saying that we don't need to log anything would be good...

Reply:
I modified the function to do the logging instead. I also changed it to clase the fileDescriptor in case `open` succeeds, but fdopen fails.